src/domains/field/compiler/resolver.ts

```diff
@@ -70,8 +71,18 @@ export function compileFieldResolver(
 
   return async (source: any, args = null, context = null, info = null) => {
     await performHooksExecution(beforeHooks, source, args, context, info);
-    const instanceField =
-      (source && source[fieldName]) || target.prototype[fieldName];
+
+    if (isGetter) {
```

Comment:
This whole block is hard to understand for me. Could we make it external function that would abstract it away with proper name. It seems like the point here is to get target property.